### PR TITLE
SPARK-4044 [CORE] Thriftserver fails to start when JAVA_HOME points to JRE instead of JDK

### PR DESCRIPTION
**Pull request by srowen**
So, I think it would be a step too far to tell people they have to run Spark with a JDK instead of a JRE. It's uncommon to put a JDK in production. So this runtime script needs to cope with not having `jar` available. 

I've added explicit checks to the two cases where it's used, but yes it's only the second datanucleus case that matters.

At least there's an error message now, but, how about just adding the jars anyway, if they're present, in this case?

CC @JoshRosen

### CODE REVIEW

File: bin/compute-classpath.sh [hunk 1]
```diff
@@ -121,10 +124,15 @@ datanucleus_jars="$(find "$datanucleus_dir" 2>/dev/null | grep "datanucleus-.*\\
...
-    echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
-    CLASSPATH="$CLASSPATH:$datanucleus_jars"
+  if [ $(command -v "$JAR_CMD") ] ; then
+    hive_files=$("$JAR_CMD" -tf "$ASSEMBLY_JAR" org/apache/hadoop/hive/ql/exec 2>/dev/null)
+    if [ -n "$hive_files" ]; then
+      echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
+      CLASSPATH="$CLASSPATH:$datanucleus_jars"
+    fi
+  else
+    echo "No jar command available; unable to check if assembly was built with Hive."
```
Comment (vanzin) [inline bin/compute-classpath.sh]: Hmm... not sure about this one. The thrift server probably won't run, in many cases, if the datanucleus jars are not added to its classpath. So won't apps that use HiveContext with a local metastore, for example. Perhaps this should be an error instead? Of maybe try to be smart and look at `$JAVA_HOME/../bin/jar` too?

(BTW, shameless plug: SPARK-4924 fixes this. But it's not in 1.3.)

Comment (srowen) [inline bin/compute-classpath.sh]: Yeah, this upgrades silence to a warning; upgrading it to an error however means that a JDK is suddenly required to run the thrift server et al, which does not seem like something that should be required (production usually disallows JDKs). However, yes, my question is: why not just always add these JARs if there's doubt, and they're present? They could be added in the `else` block, or, why bother inspecting what is built into the assembly anyway?

Comment (vanzin) [inline bin/compute-classpath.sh]: No complaints from me about always adding the jars. We could get rid of that annoying message ("Assembly was built with hive") in the process.

**Comment by srowen** [comment]
Sorry to bug @pwendell again but I think you may also be familiar with this script. I went to the extreme and removed the check for Hive jars entirely. Datanucleus goes on the classpath if it exists, full stop. This also resolves the JAR issue. But is there a reason that's a bad idea? Like, if I didn't build with Hive, but Datanucleus is lying around, does that cause a problem?

**Comment by srowen** [comment]
Ah right, this is obsoleted for `master` by SPARK-4924. I'll reopen for `branch-1.3` and possible back-porting.

**Pull request closed without merging**